### PR TITLE
Restore third party copyright headers in context package.

### LINKED ISSUE #1820: Original license headers are missing for context package.
In the PR description of #1658:

> The API is heavily inspired by gRPC and Armeria. Much code / docs are borrowed too and I kept license headers in tact to reflect that.

which is sensible and actually required by the OTel/CNCF policy:

> According to CNCF recommendations if you are contributing third-party code you will need to retain the original copyright notice.

-- https://github.com/open-telemetry/community/blob/master/CONTRIBUTING.md#copyright-notices

However, the statement from the PR description is unfortunately, wrong there is only the OTel license. Maybe the original license header got killed by spotlessApply.

CC @anuraaga

### PR DESCRIPTION
**Pull request by anuraaga**
Got lost in merge.

Fixes #1820

/cc @iNikem

### CODE REVIEW

**Review by iNikem — APPROVED** [review]

File: context/src/main/java/io/opentelemetry/context/PersistentHashArrayMappedTrie.java [hunk 1]
```diff
@@ -3,6 +3,23 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+// Includes work from:
```
Comment (Oberon00) [inline context/src/main/java/io/opentelemetry/context/PersistentHashArrayMappedTrie.java:6]:
```suggestion
// The code in this file was developed by:
```

Or something else to indicate that this file includes *solely* (mostly) work from gRPC might be better.

**Review by Oberon00 — APPROVED** [review]

File: context/src/main/java/io/opentelemetry/context/PersistentHashArrayMappedTrie.java (hunk 1 above)
Comment (iNikem) [reply context/src/main/java/io/opentelemetry/context/PersistentHashArrayMappedTrie.java:6]: What will be the criterium for "mostly"? When should we use one variant or another?

Comment (anuraaga) [reply context/src/main/java/io/opentelemetry/context/PersistentHashArrayMappedTrie.java:6]: Yeah I think it's a bit hard to judge, and we may modify the file later. In my experience, I've seen dual-license applied to files that are completely copied, not sure but probably for these reasons.

**Review by jkwatson — APPROVED** [review]
Thanks!

File: context/src/main/java/io/opentelemetry/context/PersistentHashArrayMappedTrie.java (hunk 1 above)
Comment (jkwatson) [reply context/src/main/java/io/opentelemetry/context/PersistentHashArrayMappedTrie.java:6]: I think just keeping the "Includes work from" is a good all-encompassing bit of verbiage.

**Pull request merged**